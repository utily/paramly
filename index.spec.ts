import { paramly } from "./index"

describe("paramly", () => {
	const application = new paramly.Application(
		"Test",
		"test",
		"13.37",
		[
			{
				short: "s",
				long: "server",
				arguments: 1,
				description: "Base URL of server to connect to.",
				usage: "<url>",
			},
			{
				short: "v",
				long: "verbose",
				description: "More verbose output.",
				usage: "",
			},
		],
		async f => f
	)
	application.register(
		{
			name: "output",
			description: "Show argument in output.",
			commands: {
				list: {
					name: "list",
					description: "List output.",
					examples: [
						["all", "List all output."],
						["", "List some output."],
					],
					async execute(state, argument, flags) {
						console.log("output", argument, flags, state)
						return false
					},
				},
			},
		},
		"output",
		"o"
	)
	it.each([
		"output all --server https://example.com",
		"--verbose",
		"version",
		"unknown",
		"unknown 42",
		"help",
		"help output",
		"help output list",
		"help output list --server",
		"help --server",
		"help version",
		"help unknown",
	])("%s", async argument => {
		let output = ""
		console.log = vi.fn(inputs => (output += inputs))
		expect(await application.run(["node", "test", ...argument.split(" ")])).toEqual(argument.slice(0, 6) != "output")
		expect(output).toMatchSnapshot()
	})
})
