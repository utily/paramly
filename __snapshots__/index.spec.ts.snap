// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`paramly > --verbose 1`] = `
"
Test

Usage: To get started, set a server.
The server with the name default is used by default when no --server flag is used.

 test help <module>	Get help on module

Modules:
help            Shows help.
version         Shows version.
output          Show argument in output.
"
`;

exports[`paramly > help --server 1`] = `
"
Test

Usage: To get started, set a server.
The server with the name default is used by default when no --server flag is used.

 test help <module>	Get help on module

Modules:
help            Shows help.
version         Shows version.
output          Show argument in output.
"
`;

exports[`paramly > help 1`] = `
"
Test

Usage: To get started, set a server.
The server with the name default is used by default when no --server flag is used.

 test help <module>	Get help on module

Modules:
help            Shows help.
version         Shows version.
output          Show argument in output.
"
`;

exports[`paramly > help output 1`] = `
"
Test

Usage: test output <command>		Run command
test help output <command>	Get help on command

Commands:
list      	List output.

Flags:
--server <url>               	Base URL of server to connect to.
--verbose                     	More verbose output."
`;

exports[`paramly > help output list --server 1`] = `
"
Test

Usage: test output list <command>

List output.

Examples:
all	List all output.
	List some output.
"
`;

exports[`paramly > help output list 1`] = `
"
Test

Usage: test output list <command>

List output.

Examples:
all	List all output.
	List some output.
"
`;

exports[`paramly > help unknown 1`] = `
"
Test

Usage: To get started, set a server.
The server with the name default is used by default when no --server flag is used.

 test help <module>	Get help on module

Modules:
help            Shows help.
version         Shows version.
output          Show argument in output.
"
`;

exports[`paramly > help version 1`] = `
"
Test

Usage: test version	Show version.test help version	Get help on version.
"
`;

exports[`paramly > output all --server https://example.com 1`] = `""`;

exports[`paramly > unknown 1`] = `
"
Test

Usage: To get started, set a server.
The server with the name default is used by default when no --server flag is used.

 test help <module>	Get help on module

Modules:
help            Shows help.
version         Shows version.
output          Show argument in output.
"
`;

exports[`paramly > unknown 42 1`] = `
"
Test

Usage: To get started, set a server.
The server with the name default is used by default when no --server flag is used.

 test help <module>	Get help on module

Modules:
help            Shows help.
version         Shows version.
output          Show argument in output.
"
`;

exports[`paramly > version 1`] = `
"Test
version: 13.37
"
`;
